src/client.js

```diff
@@ -228,6 +240,50 @@ class Client {
     return this.cabalToDetails(this.currentCabal)
   }
 
+  /**
+   * Add a command to the set of supported commands.
+   * @param {string} [name] the long-form command name
+   * @param {object} [cmd] the command object
+   * @param {function} [cmd.help] function returning help text
+   * @param {array} [cmd.alias] array of string aliases
+   * @param {function} [cmd.call] implementation of the command receiving (cabal, res, arg) arguments
+   */
+  addCommand(name, cmd) {
```

Comment:
what's the intent for this function? it doesn't seem to be used in cabal-client or in the cli refactor

Reply:
so you can adjust the set of commands at runtime. for things like hot-loading plugins for example